### PR TITLE
Add `ci.parameters` tag in traces for parameterized builds

### PR DESCRIPTION
**Pull request by drodriguezhdez**
### Requirements for Contributing to this repository

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix one issue at the time.
* The pull request must update the test suite to demonstrate the changed functionality.
* After you create the pull request, all status checks must be pass before a maintainer reviews your contribution. For more details, please see [CONTRIBUTING](/CONTRIBUTING.md). 

### What does this PR do?

This PR adds the`ci.parameters` tag in traces for parameterized builds for all spans.
This tag is an JSON object serialized as String.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java [hunk 1]
```diff
@@ -832,6 +832,33 @@ public static String toJson(final Set<String> set) {
...
+            return "";
+        }
+
+        // We want to avoid using Json libraries cause
+        // may cause incompatibilities on different Jenkins versions.
+        final StringBuilder sb = new StringBuilder();
+        sb.append("{");
+        int index = 1;
+        for(Map.Entry<String, String> entry : map.entrySet()) {
+            sb.append("\"").append(entry.getKey()).append("\"").append(":").append("\"").append(entry.getValue()).append("\"");
```
Comment (AdrianLC) [inline src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java:851]: nit: is there a way to do this with sprintf formatting or something like that? It'd be easier to follow

**Review by AdrianLC — DISMISSED** [review]

File: src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java [hunk 2]
```diff
@@ -832,6 +832,33 @@ public static String toJson(final Set<String> set) {
...
     }
 
+    /**
+     * Returns a JSON object string based on the map.
+     * @param map the map to transform into a JSON
+     * @return json object string
+     */
+    public static String toJson(final Map<String, String> map) {
+        if(map == null || map.isEmpty()) {
+            return "";
```
Comment (edrevo) [inline src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java:842]:
```suggestion
            return "null";
```

File: src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java (hunk 1 above)
Comment (edrevo) [reply src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java:851]: You should escape the key and value before adding them to the string. You might end up with invalid JSON if you don't do that.

File: src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java [hunk 3]
```diff
@@ -821,7 +822,8 @@ public static String toJson(final Set<String> set) {
         sb.append("[");
         int index = 1;
         for(String val : set) {
-            sb.append("\"").append(val).append("\"");
+            final String escapedValue = StringEscapeUtils.escapeJava(val);
```
Comment (edrevo) [inline src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java:825]:
```suggestion
            final String escapedValue = StringEscapeUtils.escapeJavaScript(val);
```

**Review by edrevo — APPROVED** [review]

**Pull request merged**